### PR TITLE
Fix modified shape function gradients type

### PR DESCRIPTION
**Pull request by jcotela**
Changing the interface to use the same types for shape function derivatives as the geometry.

### CODE REVIEW

File: kratos/modified_shape_functions/modified_shape_functions.cpp [hunk 1]
```diff
@@ -292,7 +294,7 @@ namespace Kratos
                 }
 
                 aux_grad_sh_func_cond = prod(aux_grad_sh_func_exp, rPmatrix);
-                rInterfaceShapeFunctionsGradientsValues.push_back(trans(aux_grad_sh_func_cond));
+                rInterfaceShapeFunctionsGradientsValues[i_interface*n_int_pts + i_gauss] = trans(aux_grad_sh_func_cond);
```
Comment (jcotela) [inline kratos/modified_shape_functions/modified_shape_functions.cpp:297]: @rubenzorrilla please check that I put the right index here.

Comment (rubenzorrilla) [reply kratos/modified_shape_functions/modified_shape_functions.cpp:297]: It's correct! :1st_place_medal:

Comment (rubenzorrilla) [reply kratos/modified_shape_functions/modified_shape_functions.cpp]: Yes, it's the right one.

File: kratos/modified_shape_functions/tetrahedra_3d_4_modified_shape_functions.cpp [hunk 2]
```diff
@@ -65,7 +65,7 @@ bool Tetrahedra3D4ModifiedShapeFunctions::IsSplit() {
 // Internally computes the splitting pattern and returns all the shape function values for the positive side.
 void Tetrahedra3D4ModifiedShapeFunctions::ComputePositiveSideShapeFunctionsAndGradientsValues(
     Matrix &rPositiveSideShapeFunctionsValues,
-    std::vector<Matrix> &rPositiveSideShapeFunctionsGradientsValues,
+    ModifiedShapeFunctions::ShapeFunctionsGradientsType &rPositiveSideShapeFunctionsGradientsValues,
```
Comment (rubenzorrilla) [inline kratos/modified_shape_functions/tetrahedra_3d_4_modified_shape_functions.cpp]: I think you can remove the ModifiedShapeFunctions::

**Review by rubenzorrilla — APPROVED** [review]

**Pull request merged**